### ISSUE TITLE
Trivial permutation and permutation groups

### ISSUE THREAD

**Issue by fafcb8fb-f679-4d5a-b5d7-a3b49c3076ce** [sagemath/sage#4521]
<div id="comment:0"></div>

When constructing an empty permutation the degree should be zero (ie a permutation of the empty set). But

```
sage: PermutationGroupElement([]).parent().domain()
{1}
```
Incidentally, it is also wrong for `PermutationGroup`

```
sage: G = PermutationGroup([])
sage: G.degree()
1
```

Maybe `PermutationGroupElement` and `PermutationGroup` should take an argument for the degree, e.g.:

```
sage: G = PermutationGroup([],degree=4)
```

NOTE: one (not compltely ideal) workaround is to use `SymmetricGroup(4).subgroup([])`

---

The original report mentioned two other problems that now behave coherently (tested on 9.0.beta3)

1. This gives an error:

```
sage: G = PermutationGroup([])
sage: G.list()
```

4. Degree 0 should really be supported or 
we will have difficulties with automorphism 
groups of boundary cases.  Currently this 
gives an error:

```
sage: SymmetricGroup(0)
```
Certainly these examples should go into the  docstrings. Most of these can be trivially fixed, but it would be good if someone could review permutation groups with a view to catching these problems before they arise.

Depends on #28652

CC:  @saliola @mwhansen @KPanComputes

Component: **group theory**

_Issue created by migration from https://trac.sagemath.org/ticket/4521_

**Comment by videlec** [comment]
Seems fixed in 10.0.beta3
```
sage:  PermutationGroupElement([]).parent().domain()
{}
```

**Comment by dcoudert** [comment]
Let's close this issue.
